lib/auth/auth_with_roles.go

```diff
@@ -6409,11 +6409,32 @@ func (a *ServerWithRoles) GetWindowsDesktops(ctx context.Context, filter types.W
 	return filtered, nil
 }
 
+// hostIDExistsAndIsValid checks if the host ID exists and matches a registered windows desktop service.
+func (a *ServerWithRoles) hostIDExistsAndIsValid(ctx context.Context, s types.WindowsDesktop) error {
+	hostID := s.GetHostID()
+	if hostID == "" {
+		return trace.BadParameter("host_id is required")
+	}
+	wdss, err := a.authServer.GetWindowsDesktopServices(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, wds := range wdss {
+		if wds.GetName() == hostID {
+			return nil
+		}
+	}
```

Comment:
Why look up all desktop services (there may be thousands of them) rather than use `GetWindowsDesktopService` and looking up a single w_d_s by name?

Reply:
derp: 106c26f4b916c4108b57be8b9f5f84974f8c595d